pkgs/by-name/od/odin/package.nix

```diff
@@ -24,16 +21,12 @@ stdenv.mkDerivation {
     hash = "sha256-GXea4+OIFyAhTqmDh2q+ewTUqI92ikOsa2s83UH2r58=";
   };
 
-  postPatch =
-    lib.optionalString stdenv.hostPlatform.isDarwin ''
-      substituteInPlace src/linker.cpp \
-          --replace-fail '/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk' ${MacOSX-SDK}
-    ''
-    + ''
-      substituteInPlace build_odin.sh \
-          --replace-fail '-framework System' '-lSystem'
-      patchShebangs build_odin.sh
-    '';
+  postPatch = lib.optionalString stdenv.hostPlatform.isDarwin ''
+    substituteInPlace src/linker.cpp \
+        --replace-fail '-Wl,-syslibroot /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk -L/usr/local/lib ' "" \
+        --replace-fail '-L/opt/homebrew/lib ' "" \
+        --replace-fail '-L/opt/local/lib ' ""
```

Comment:
nit: I would prefer a proper `.patch` that removes these lines entirely, to not have pointless runtime checks left in and to catch it at build time if they think of even more system paths to add.